src/components/ManaCalculator/hooks/useManaState.ts

```diff
@@ -162,7 +171,27 @@ export function useGivenManaState(
   }
 
   function handleOwnHoldChange(value: number) {
-    setState({ ...state, heldTokens: value });
+    setState({
+      ...state,
+      ...getDerivedValuesFromHeldTokens(value),
+    });
+  }
+
+  function getDerivedValuesFromHeldTokens(
+    heldTokens: number,
+  ): Partial<ManaCalculatorProps> {
+    return {
+      heldTokens,
+      [getStakedOrDelegated(state.userType)]: heldTokens,
```

Comment:
why is this new key pair needed?

Reply:
when the held tokens change we should set the same in staked/delegated tokens, right? It's what the issue says

Reply:
Ah sorry didnt understand what the function meant 🙏🏼 so that returns the stake or deleted amount key, gotit, resolving